mesonbuild/backend/backends.py

```diff
@@ -155,17 +155,22 @@ def get_target_dir(self, target):
             dirname = 'meson-out'
         return dirname
 
+    @staticmethod
+    def path_join(a, b):
+        # helper to join b to a if b is not empty.
+        # avoids adding an extraneous trailing os.path.sep
+        if b:
+            return os.path.join(a, b)
+        return a
+
     def get_target_source_dir(self, target):
-        dirname = os.path.join(self.build_to_src, self.get_target_dir(target))
-        return dirname
+        return Backend.path_join(self.build_to_src, self.get_target_dir(target))
 
     def get_target_private_dir(self, target):
-        dirname = os.path.join(self.get_target_dir(target), target.get_basename() + target.type_suffix())
-        return dirname
+        return Backend.path_join(self.get_target_dir(target), target.get_basename() + target.type_suffix())
```

Comment:
The second argument will never be empty in this case. The first argument might be, but `os.path.join` handles that just fine.